### DISCUSSION TITLE
Applying site-wise rates of evolution to alisim

### DISCUSSION THREAD

**Discussion by ElizabethRobbins** [iqtree/iqtree2#231] [Q&A] [answered]
Hello,

Is there a way of applying site-wise rates of evolution calculated by IQ-TREE to the alisim function to make the simulated alignment more realistic? For example, if a site experiences no variation across the 'real' alignment, it remains unchanged in all the simulated alignments.

Many thanks,
Lizzie

**Comment by trongnhanuit** [comment, marked as answer]
Hi Lizzie,
Thank you for using AliSim and for your question.
In AliSim, we developed a feature to simulate new alignments that mimic a
real alignment. In fact, we internally run IQ-TREE to infer the model
parameters and the tree, and then AliSim simulates new alignments using
that inferred information. I think that feature is what you're looking for.
For site-wise rates, we offer different options to apply prior and
posterior distributions inferred from the real alignment. For more details,
please have a look at
http://www.iqtree.org/doc/AliSim#mimicking-a-real-alignment.

Thank you.

Cheers,
Nhan

**Reply by bqminh** [reply]
@trongnhanuit The question was, whether you can take `.rate` file (produced by IQ-TREE via `--rate` option) as input for AliSim?

**Comment by trongnhanuit** [comment]
Hi Lizzie,
AliSim doesn't take the *.rate file as input. However, you can consider my
suggestion in the previous response. Hope it helpful!

Cheers,
Nhan

**Comment by ElizabethRobbins** [comment]
Thank you both for your replies. The reason I ask is because I am simulating an alignment for a protein that has ~75% of its sites that are invariant across a dataset of ~800 species using a root sequence. Therefore, I want to direct the amino acid changes to variant regions of the protein. Furthermore, the simulation would be improved if the variant sites were put into buckets e.g., slow or fast evolving would allow a more realistic simulation of the protein to be produced while still allowing variation between simulations. Would something like this be a possible additional feature of Alisim? Secondly, with the current version of iqtree, how would you go about simulating the most realistic alignment from a root sequence? (E.g., I have thought of providing AliSim with a root sequence where the invariant sites are removed and adjusting the branch lengths of the tree accordingly to eliminate variation in invariant regions of the protein). 

Many thanks,
Lizzie

**Comment by roblanf** [comment]
Hi Lizzie,

As far as I can tell you can already achieve all of this with AliSim. For example, if you have 75% invariant sites, plus a range of rates of evolution, you would:

1. Estimate the rate distribution with ModelFinder. Let's say for arguments sake you get +I+R5
2. Simulate alignments with AliSim under that model

This should give you very similar rate heterogeneity across sites to your original alignment. I note that in phylogenetics the _order_ of the sites in the alignment doesn't matter, and so this is not accounted for in the output, i.e. the fast and slow sites will be in different _positions_ in the alignment compared to your input file, but they should have the same underlying distribution. 

Another thing you can do if you want to control the ordering of sites is simulate one alignment for each rate category, then concatenate them in the order and proportions that you want. This is how people used to simulate data under e.g. the +G models. 

Rob
